docs/pages/release_notes.md

```diff
@@ -50,42 +50,96 @@ This means, you can use CPD to find duplicated code in your Kotlin projects.
+
+```
+
+You're highly encouraged to migrate to using this new API as soon as possible, to ease your migration to 7.0.0.
+
+
+
+##### Architectural simplifications
+
+* {% jdoc props::EnumeratedPropertyDescriptor %}, {% jdoc props::NumericPropertyDescriptor %}, {% jdoc props::PackagedPropertyDescriptor %},
+and the related builders (in {% jdoc_package props::builders %}) will be removed.
+These specialized interfaces allowed additionnal constraints to be enforced on the
+value of a property, but made the property class hierarchy very large and impractical
+to maintain. Their functionality will be mapped uniformly to {% jdoc props::constraints.PropertyConstraint %}s,
+which will allow virtually any constraint to be defined, and improve documentation and error reporting. The
+related methods {% jdoc !c!props::PropertyTypeId#isPropertyNumeric() %} and
+{% jdoc !c!props::PropertyTypeId#isPropertyPackaged() %} are also deprecated.
+
+* {% jdoc props::MultiValuePropertyDescriptor %} and {% jdoc props::SingleValuePropertyDescriptor %}
+are deprecated. 7.0.0 will introduce a new XML syntax which will remove the need for such a divide
+between single- and multi-valued properties. The method {% jdoc !c!:PDr#isMultiValue() %} will be removed
+accordingly.
+
+##### Changes to the PropertyDescriptor interface
+
+* {% jdoc :PDr#preferredRowCount() %} is deprecated with no intended replacement. It was never implemented, and does not belong
+  in this interface. The methods {% jdoc :PDr#uiOrder() %} and `compareTo(PropertyDescriptor)` are deprecated for the
+  same reason. These methods mix presentation logic with business logic and are not necessary for PropertyDescriptors to work.
+  `PropertyDescriptor` will not extend `Comparable<PropertyDescriptor>` anymore come 7.0.0.
+* The method {% jdoc :PDr#propertyErrorFor(core::Rule) %} is deprecated and will be removed with no intended
+  replacement. It's really just a shortcut for `prop.errorFor(rule.getProperty(prop))`.
+* `T `{% jdoc !a!:PDr#valueFrom(java.lang.String) %} and `String `{% jdoc :PDr#asDelimitedString(java.lang.Object) %}`(T)` are deprecated and will be removed. These were
+  used to serialize and deserialize properties to/from a string, but 7.0.0 will introduce a more flexible
+  XML syntax which will make them obsolete.
+* {% jdoc :PDr#isMultiValue() %} and {% jdoc :PDr#type() %} are deprecated and won't be replaced. The new XML syntax will remove the need
+  for a divide between multi- and single-value properties, and will allow arbitrary types to be represented.
+  Since arbitrary types may be represented, `type` will become obsolete as it can't represent generic types,
+  which will nevertheless be representable with the XML syntax. It was only used for documentation, but a
+  new way to document these properties exhaustively will be added with 7.0.0.
+* {% jdoc :PDr#errorFor(java.lang.Object) %} is deprecated as its return type will be changed to `Optional<String>` with the shift to Java 8.
```

Comment:
This might be different - if we collect errors and return all violated constraints on the property, we might just add a `List<String> errorsFor(T value)` or something even more sophisticated. We could have a look e.g. at the [bean validation api](https://docs.oracle.com/javaee/7/api/javax/validation/Validator.html) - there a `Set<ConstraintValidation>` is returned. So we could return something more structured than just a string.